src/simularium/WebsocketClient.ts

```diff
@@ -313,7 +316,7 @@ export class WebsocketClient {
         jsonData: Record<string, unknown>,
         requestDescription: string
     ): void {
-        if (this.socketIsValid()) {
+        if (this.socketIsConnected()) {
```

Comment:
I ran into this issue when testing where the socket can be "valid" (which means the websocket's state isn't CLOSED), but not "connected" (which means the websocket's state is OPEN) if the state is PENDING, in which case sending a message throws an error. I feel like this check is trying to prevent that error, so might as well cover the PENDING case too?